src/backend/core/warren/base_indicator.py

```diff
@@ -2,9 +2,59 @@
 from abc import ABC, abstractmethod
 from typing import List
 
+import pandas as pd
+import numpy as np
 from ralph.backends.http.lrs import LRSQuery
 
 
+def parse_raw_statements(raw_statements) -> pd.DataFrame:
+    """Parse LRS statements, explode the columns, and add a `date` column."""
+    flattened = pd.json_normalize(raw_statements)
+    # Disable chained assignment warning to make the transformation inplace
+    pd.options.mode.chained_assignment = None
+    # Transform timestamp column into a date with day (YYYY-MM-DD)
+    flattened.loc[:, "date"] = pd.to_datetime(flattened.loc[:, "timestamp"]).dt.date
+    df = add_unique_actor_id(flattened)
+    return df
+
+
+def add_unique_actor_id(statements: pd.DataFrame) -> pd.DataFrame:
+    """Depending on the xAPI statements, the actor can be identified in 4 different ways :
+    https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#details-4
+
+    This function creates a `unique_agent_id` column that uniquely identifies the agent.
+    """
+
+    columns_to_check = ["actor.account.name", "actor.account.homePage", "actor.mbox", "actor.openid"]
```

Comment:
shouldn't it a be constant set? 
```suggestion
    COLUMNS_TO_CHECK = {"actor.account.name", "actor.account.homePage", "actor.mbox", "actor.openid"}
```

Reply:
Yes, I agree. Pycharm gives a warning when defining a constant inside a function, but I think we should ignore it and follow the convention.